dev/test/transaction.ts

```diff
@@ -416,14 +424,51 @@ describe('failed transactions', () => {
     });
   });
 
-  it("doesn't retry on callback failure", () => {
+  it('retries GRPC exceptions with code ABORTED in callback', () => {
+    const retryableError = new GoogleError('Aborted');
+    retryableError.code = Status.ABORTED;
+
+    return runTransaction(
+      async (transaction, docRef) => {
+        await transaction.get(docRef);
+        return 'success';
+      },
+      begin('foo1'),
+      getDocument('foo1', retryableError),
+      rollback('foo1'),
+      begin('foo2', 'foo1'),
+      getDocument('foo2'),
+      commit('foo2')
+    ).then(red => {
+      expect(red).to.equal('success');
+    });
+  });
+
+  it("doesn't retry GRPC exceptions with code FAILED_PRECONDITION in callback", () => {
+    const nonRetryableError = new GoogleError('Failed Precondition');
+    nonRetryableError.code = Status.FAILED_PRECONDITION;
+
+    return expect(
+      runTransaction(
+        async (transaction, docRef) => {
+          await transaction.get(docRef);
+          return 'failure';
+        },
+        begin('foo'),
+        getDocument('foo', nonRetryableError),
+        rollback('foo')
+      )
+    ).to.eventually.be.rejectedWith('Failed Precondition');
+  });
+
+  it("doesn't retry custom user exceptions in callback", () => {
     return expect(
       runTransaction(
         () => {
           return Promise.reject('request exception');
         },
         begin(),
-        rollback('foo')
+        rollback()
```

Comment:
should we make it consistent throughout [elsewhere](https://github.com/googleapis/nodejs-firestore/blob/8f41b37fd766b18113401c7324e319b52b38cc96/dev/test/transaction.ts#L399-L400)

Reply:
Fixed